stdlib/asm/crypto/dsa/ecdsa/secp256k1.masm

```diff
@@ -1,285 +1,105 @@
-
-    locaddr.20
-    locaddr.16
-    locaddr.12
-    locaddr.8
-    locaddr.4
-    locaddr.0
-
-    # compute pkey * ((r * s^-1) mod N) = P1
-    exec.group::mul
-
-    push.0.0.0.0
+# ENCODING CONVENTIONS
+# ================================================================================================
+#
+# This module uses the following notation for data representation:
+# - Byte arrays are stored in memory as packed u32 values in little-endian format
+# - Each u32 represents 4 bytes: u32 = u32::from_le_bytes([b0, b1, b2, b3])
+# - Unused bytes in the final u32 must be set to zero
+# - Memory addresses must be word-aligned (divisible by 4)
 
-    movup.4
-    mem_loadw_be
-    loc_storew_be.0
+# Constants
+# ================================================================================================
 
-    movup.4
-    mem_loadw_be
-    loc_storew_be.4
+const.ECDSA_VERIFY_EVENT=event("stdlib::crypto::dsa::ecdsa::verify")
 
-    movup.4
-    mem_loadw_be
-    loc_storew_be.8
+const.PK_LEN_FELTS=9     # 33.div_ceil(4)
+const.DIGEST_LEN_FELTS=8 # 32.div_ceil(4)
+const.SIG_LEN_FELTS=17   # 66.div_ceil(4)
 
-    movup.4
-    mem_loadw_be
-    loc_storew_be.12
+# PROCEDURES
+# ================================================================================================
 
-    movup.4
-    mem_loadw_be
-    loc_storew_be.16
-
-    movup.4
-    mem_loadw_be
-    loc_storew_be.20
-
-    dropw
-
-    locaddr.92
-    locaddr.88
-    locaddr.84
-    locaddr.80
-    locaddr.76
-    locaddr.72
-
-    locaddr.68
-    locaddr.64
-    locaddr.60
-    locaddr.56
-    locaddr.52
-    locaddr.48
-
-    locaddr.20
-    locaddr.16
-    locaddr.12
-    locaddr.8
-    locaddr.4
-    locaddr.0
-
-    # compute P0 + P1 = P2
-    exec.group::add
-
-    push.0.0.0.0
-
-    movup.4
-    mem_loadw_be
-    loc_storew_be.0
-
-    movup.4
-    mem_loadw_be
-    loc_storew_be.4
-
-    movup.4
-    mem_loadw_be
-    loc_storew_be.8
-
-    movup.4
-    mem_loadw_be
-    loc_storew_be.12
-
-    movup.4
-    mem_loadw_be
-    loc_storew_be.16
-
-    movup.4
-    mem_loadw_be
-    loc_storew_be.20
-
-    push.0.0.0.0
-    loc_loadw_be.20
-    swapw
-    loc_loadw_be.16
-
-    exec.base_field::inv
-
-    push.0.0.0.0.0.0.0.0
-    loc_loadw_be.4
-    swapw
-    loc_loadw_be.0
-
-    # compute r' ( in radix-2^32 form )
-    exec.base_field::mul
-    exec.base_field::from_mont
-
-    push.0.0.0.0.0.0.0.0
-    loc_loadw_be.36
-    swapw
-    loc_loadw_be.32
-
-    # compute r ( in radix-2^32 form )
-    exec.scalar_field::from_mont
-
-    # check if r == r'
-    movup.8
-    assert_eq
-
-    movup.7
-    assert_eq
-
-    movup.6
-    assert_eq
-
-    movup.5
-    assert_eq
-
-    movup.4
-    assert_eq
-
-    movup.3
-    assert_eq
+#! Verifies an ECDSA signature over the secp256k1 curve.
+#!
+#! This procedure uses deferred verification via a precompile. The actual cryptographic
+#! verification is performed by the host, and the result is provided via the advice stack.
+#!
```

Comment:
I would maybe mention somewhere here that this procedure is agnostic as to what hash algorithm was used to generate the message digest.